addons/sms/tests/common.py

```diff
@@ -237,6 +237,22 @@ def assertSMSNotification(self, recipients_info, content, messages=None, check_s
         self.assertEqual(notifications.mapped('res_partner_id'), partners)
 
         for recipient_info in recipients_info:
+            # sanity check
+            extra_keys = set(recipient_info.keys()) - {
```

Comment:
Works too

```suggestion
            extra_keys = recipient_info.keys() - {
```

And you'll earn space and improve readability by limiting the intended scope of `extra_keys` with
```suggestion
            if extra_keys := recipient_info.keys() - {
                ...,
            }:
                raise ...
```

Reply:
:= is "recent", not sure it is ok for 17. Ok for extra set to remove.